lib/src/builder_implementation.dart

```diff
@@ -4534,13 +4513,43 @@ String _extractMetadataCode(Element element, Resolver resolver,
       if (_isPrivateName(annotationNode.name.name)) {
         _severe("Cannot access private name ${annotationNode.name}", element);
       }
-      metadataParts.add("$prefix${annotationNode.name}");
+      String name = _extractNameWithoutPrefix(annotationNode.name, element);
+      if (_isPrivateName(name)) {
+        _severe("Cannot access private name $name", element);
```

Comment:
Not specific to this change, but it would be great if we could output a source location of where name is accessed.

Reply:
This is already in place: We pass `element` as the `errorTarget` which is used to print a source code location, and that gets passed on to `_formatDiagnosticMessage` where the source code location is extracted and included in the returned string.